src/components/Editor/CitationManager/index.js

```diff
@@ -0,0 +1,333 @@
+import store from '@/store/index'
+
+class CitationOpError extends Error {
+  /**
+   * Custom error for operation on citation
+   * @param {String} msg message of error
+   * @param  {...any} params other parameters
+   */
+  constructor(msg, ...params) {
+    super(msg, ...params)
+
+    // Maintains stack trace (V8 only)
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, CitationOpError)
+    }
+
+    this.name = 'CitationOpError'
+  }
+}
+
+class CitationNode {
+  /**
+   * constructor of CitationNode
+   * @param {Node} node
+   */
+  constructor(node) {
+    this.node = node
+  }
+
+  /**
+   * get id of the editor that owns this node
+   */
+  get editorId() {
+    return this.node.$props.editor.view.dom.id
+  }
+
+  /**
+   * get pos of the node in the editor
+   */
+  get pos() {
+    return this.node.getPos()
+  }
+}
+
+class CitationInfo {
+  /**
+   * constructor
+   * @param {String} title : title of citation
+   * @param {String} url : url of citation
+   */
+  constructor(title, url) {
+    new URL(url)
```

Comment:
Does this use or error handling?

Reply:
This is only for error handling, but the `CitationManager` is not responsible for catching error. The error will be handled in components.